### ISSUE TITLE
Libtorchcodec error.

### ISSUE THREAD

**Issue by marius-sucan** [yuvraj108c/ComfyUI-Whisper#37]
I get this error:

Procedure ?decref_pyobject@TensorImpl@c10@@UEBAXXZ not found in DLL: C:\ComfyUI\python_embeded\Lib\site-packages\torchcodec\libtorchcodec_core4.dll. 

What version does it require?

Thank you.

**Comment by marius-sucan** [comment]
The solution is... It needs torch codec 0.9.0 and ffmpeg essentials. You can place it in the python_embed folder.

**Comment by trollver9000** [comment]
NO! 
This is solution , open apply whisper.py and replace 


    def apply_whisper(self, audio, model, language, prompt):
        # --- Extract and preprocess audio tensor ---
        waveform = audio['waveform']  # Shape: [1, C, T] or [1, T]
        sample_rate = audio["sample_rate"]

        # Remove batch dimension -> [C, T]
        if waveform.dim() == 3:
            waveform = waveform.squeeze(0)  # [1, C, T] -> [C, T]
        elif waveform.dim() == 2:
            pass  # Already [C, T]
        else:
            raise ValueError(f"Unexpected waveform shape: {waveform.shape}")

        # Convert to mono -> [T]
        if waveform.shape[0] > 1:
            waveform = torch.mean(waveform, dim=0)  # [C, T] -> [T]
        else:
            waveform = waveform.squeeze(0)  # [1, T] -> [T]

        # Resample to 16kHz if needed
        if sample_rate != 16000:
            import torchaudio
            resampler = torchaudio.transforms.Resample(orig_freq=sample_rate, new_freq=16000)
            waveform = resampler(waveform)

        # Convert to NumPy (MUST be 1D array)
        audio_array = waveform.cpu().numpy()  # Shape: [T]

        # --- Rest of your code ---
        cache_key = model
        if cache_key not in WHISPER_PATCHER_CACHE:
            load_device = mm.get_torch_device()
            download_root = os.path.join(folder_paths.models_dir, WHISPER_MODEL_SUBDIR)
            logger.info(f"Creating Whisper ModelPatcher for {model} on device {load_device}")
            
            model_wrapper = WhisperModelWrapper(model, download_root)
            patcher = WhisperPatcher(
                model=model_wrapper,
                load_device=load_device,
                offload_device=mm.unet_offload_device(),
                size=0
            )
            WHISPER_PATCHER_CACHE[cache_key] = patcher

        patcher = WHISPER_PATCHER_CACHE[cache_key]
        mm.load_model_gpu(patcher)
        whisper_model = patcher.model.whisper_model

        if whisper_model is None:
            logger.error("Whisper model failed to load.")
            raise RuntimeError(f"Failed to load Whisper model: {model}")

        transcribe_args = {"initial_prompt": prompt}
        if language != "auto":
            if ApplyWhisperNode.languages_by_name is None:
                ApplyWhisperNode.languages_by_name = {v.lower(): k for k, v in whisper.tokenizer.LANGUAGES.items()}
            transcribe_args['language'] = ApplyWhisperNode.languages_by_name[language.lower()]
        
        # ✅ CORRECT CALL: 1D ARRAY, NO sample_rate
        result = whisper_model.transcribe(
            audio_array,
            word_timestamps=True,
            **transcribe_args
        )

        # --- Build alignments ---
        segments = result['segments']
        segments_alignment = []
        words_alignment = []

        for segment in segments:
            segment_dict = {
                'value': segment['text'].strip(),
                'start': segment['start'],
                'end': segment['end']
            }
            segments_alignment.append(segment_dict)

            for word in segment["words"]:
                word_dict = {
                    'value': word["word"].strip(),
                    'start': word["start"],
                    'end': word['end']
                }
                words_alignment.append(word_dict)

        return (result["text"].strip(), segments_alignment, words_alignment)

**Comment by marius-sucan** [comment]
Thank you . I have not tested it, but I suppose your solution is a better one.

My proposal fixes two problems: 
1. The incompatibility with the lib torchcodec.
2. Not finding FFmpeg.

**Comment by yuvraj108c** [comment]
Latest version uses soundfile and should resolve the torchcodec issue
Plz update
